modules/ingester/instance.go

```diff
@@ -61,6 +63,8 @@ type instance struct {
 	bytesWrittenTotal  prometheus.Counter
 	limiter            *Limiter
 	wal                *tempodb_wal.WAL
+
+	hash hash.Hash32
```

Comment:
it occurs to me that this might not be concurrency safe.  if it's not we can't make this change.  apologies for sending you down the wrong path.

obviously the `CombineTraces` change is still fine b/c it is all executed in the same goroutine.

Reply:
Yes that was considered, but it seems like it is concurrency safe?   The only call chain is under the traces mutex lock via tokenForTraceID <- getOrCreateTrace <- Push.

Reply:
Yup, you are correct.  It is protected by a mutex.